python/python/bystro/proteomics/annotation_interface.py

```diff
@@ -48,93 +42,182 @@ def _flatten(xs: Any) -> list[Any]:  # noqa: ANN401 (`Any` is really correct her
+            lambda x: x
+            not in [
+                "chrom",
+                "pos",
+                "id",
+                "vcfPos",
+                "inputRef",
+                "ref",
+                "alt",
+                "refSeq.name2",
+                "refSeq.name",
+                "refSeq.protAcc",
+                "refSeq.isCanonical",
+                "heterozygotes",
+                "homozygotes",
+                "missingGenos",
+                "heterozygosity",
+                "homozygosity",
+                "missingness",
+                "gnomad.genomes.AF",
+                "gnomad.exomes.AF",
+            ],
+            additional_fields if additional_fields is not None else [],
+        )
+    )
+
     rows = []
-    for gene_name in unique_gene_names:
-        for heterozygote in heterozygotes:
-            rows.append(
-                {
-                    "sample_id": heterozygote,
-                    "chrom": chrom,
-                    "pos": pos,
-                    "ref": ref,
-                    "alt": alt,
-                    "gene_name": gene_name,
-                    "dosage": HETEROZYGOTE_DOSAGE,
-                }
-            )
-        for homozygote in homozygotes:
-            rows.append(
-                {
-                    "sample_id": homozygote,
-                    "chrom": chrom,
-                    "pos": pos,
-                    "ref": ref,
-                    "alt": alt,
-                    "gene_name": gene_name,
-                    "dosage": HOMOZYGOTE_DOSAGE,
-                }
-            )
-        for missing_geno in missing_genos:
-            rows.append(
-                {
-                    "sample_id": missing_geno,
+    for gene_idx, gene_name in enumerate(gene_names):
+        for sample_list, dosage_label in [
+            (heterozygotes, 1),
+            (homozygotes, 2),
+            (missing_genos, -1),
+        ]:
+            for sample_id in sample_list:
+                row = {
+                    "sample_id": sample_id,
                     "chrom": chrom,
+                    "vcf_pos": vcf_pos,
                     "pos": pos,
+                    "id": _id,
                     "ref": ref,
+                    "input_ref": input_ref,
                     "alt": alt,
                     "gene_name": gene_name,
-                    "dosage": MISSING_GENO_DOSAGE,
+                    "transcript_name": transcript_names[gene_idx],
+                    "protein_name": protein_names[gene_idx],
+                    "is_canonical": is_canonical[gene_idx],
+                    "dosage": dosage_label,
+                    "heterozygosity": heterozygosity,
+                    "homozygosity": homozygosity,
+                    "missingness": missingness,
+                    "gnomad_genomes_af": gnomad_genomes_af,
+                    "gnomad_exomes_af": gnomad_exomes_af,
                 }
-            )
+                # Add additional fields
```

Comment:
More of a stylistic question, but is there a reason why you wouldn't want to handle all fields using the code below instead of specifying each one of the defaults? Is it so that you can change the name of the fields to something friendlier to read/interpret or a different reason?

Reply:
We could specify a list of default "fields to add"; we just wouldn't be able to rename them as we do above. I'm not committed to the renaming; do you think it is useful?